streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsMetricsImplTest.java

```diff
@@ -53,19 +57,27 @@ public void testRemoveSensor() {
         String entity = "entity";
         String operation = "put";
         Map<String, String> tags = new HashMap<>();
-        StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(new Metrics(), groupName, tags);
+        final Metrics metrics = new Metrics();
+        final Map<MetricName, KafkaMetric> initialMetrics = Collections.unmodifiableMap(new LinkedHashMap<>(metrics.metrics()));
+        StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, groupName, tags);
 
         Sensor sensor1 = streamsMetrics.addSensor(sensorName, Sensor.RecordingLevel.DEBUG);
         streamsMetrics.removeSensor(sensor1);
+        Assert.assertEquals(initialMetrics, metrics.metrics());
```

Comment:
What are those assertions validated for?

Reply:
I'm checking that the metric actually got removed from the registry. Previously, we didn't assert anything, which I guess validates that we didn't get any exceptions, but if we failed to remove the sensor, the test would still pass.